R/matching.R

```diff
@@ -214,7 +210,7 @@ plot_weights <- function(wt, bin_col = "#6ECEB2", vline_col = "#688CE8", main_ti
 #' )
 #'
 #' print(check)
-check_weights <- function(optimized, processed_agd) {
+check_weights <- function(match_res, processed_agd) {
```

Comment:
`match_res` isn't totally obvious either. Would `weighted_ipd` be clearer?

Reply:
@gravesti 
weighted_ipd is also kind of confusing since what is returned is a list containing many components. Do you have any other suggestion?
 
# Output
  list(
    data = data,
    centered_colnames = centered_colnames,
    nr_missing = nr_missing,
    ess = sum(wt)^2 / sum(wt^2),
    opt = opt1
  )

Reply:
You could also just have the argument as `object`. My idea for `weighted_ipd` follows from the function calls.
```r
target_pop <- process_agd(target_pop)
adsl %>%
 center_ipd(agd = target_pop) %>%
 estimate_weights() %>%
 check_weights(??? = ., processed_agd = target_pop)
```
In the same way that `processed_agd` comes from `process_agd()`, we have some centred and weighted IPD.
I don't think it's a problem that it is actually a list of components. They are all related to the weighting.

Maybe `weighted_data`? since `estimate_weights()` has a `data=` parameter.

Reply:
Yea I think weighted_data is better. Thanks!